### PR TITLE
Allow custom headers for request

### PR DESCRIPTION
**Pull request by chl03ks**
Adding `marathonHeaders` to allow custom headers for the request this allow us to make request to the api outside the cluster or in development

### CODE REVIEW

File: lib/MarathonEventBusClient.js [hunk 1]
```diff
@@ -101,7 +108,9 @@ MarathonEventBusClient.prototype.subscribe = function () {
         url = self.options.marathonProtocol + "://" + self.options.marathonHost + ":" + self.options.marathonPort+self.options.marathonUri;
 
     // Allow unauthorized HTTPS requests
-    let eventSourceInitDict = { rejectUnauthorized: false };
+    let eventSourceInitDict = self.options.marathonHeaders;
```
Comment (tobilg) [inline lib/MarathonEventBusClient.js]: Not sure if this is a good idea here... First, the headers need to be in a `headers` property (See https://www.npmjs.com/package/eventsource#setting-http-request-headers), which I don't see in the code (usage example missing?). 

Furthermore, I'd like it better if you would have a whitelist of allowed header keys, and check if the `marathonHeaders` contains valid headers, and only using the ones which pass the whitelist.

**Review by tobilg — COMMENTED** [review]
Thanks for the PR! I have some small remarks, please have a look at the review comment.

File: lib/MarathonEventBusClient.js (hunk 1 above)
Comment (chl03ks) [reply lib/MarathonEventBusClient.js]: cool that sounds good and yeah maybe didnt wrote my code right at it does not looks that i have th header property on the object but i'll change that in the next commit.

**Comment by tuxity** [comment]
I'm not sure but, it should allow headers yes, but these headers in the code come from DC/OS not Marathon. That's 2 different things you have to keep in mind.
See https://dcos.io/docs/1.8/administration/id-and-access-mgt/iam-api/ DC/OS protect Marathon API from requests made outside of the cluster where Marathon is

**Comment by chl03ks** [comment]
@tuxity  @tobilg  At the time i just allow 'Authorization' and 'Content-type' on the 'marathonHeaders' property i don't know if there should be more allowedHeaders on the list

File: lib/MarathonEventBusClient.js [hunk 2]
```diff
@@ -21,7 +21,13 @@ function MarathonEventBusClient (options) {
...
+    self.allowedEventTypes = [
+      "pod_created_event","pod_updated_event", "pod_deleted_event","scheduler_registered_event", "scheduler_reregistered_event", "scheduler_disconnected_event", "subscribe_event", "unsubscribe_event", "event_stream_attached", "event_stream_detached", "add_health_check_event", "remove_health_check_event", "failed_health_check_event", "health_status_changed_event", "unhealthy_task_kill_event", "group_change_success", "group_change_failed", "deployment_info", "deployment_success", "deployment_failed", "deployment_step_success", "deployment_step_failure", "app_terminated_event", "status_update_event", "instance_changed_event", "unknown_instance_terminated_event", "instance_health_changed_event", "framework_message_event"
+    ];
+
+    self.allowedHeaeders = [
```
Comment (tuxity) [inline lib/MarathonEventBusClient.js]: `self.allowedHeaders`

File: README.md [hunk 3]
```diff
@@ -63,8 +63,10 @@ You can specify the following properties when instantiating the Marathon Event B
 
  * `marathonHost`: The Marathon base URL. Default is `master.mesos`.
  * `marathonPort`: The Marathon port. Default is `8080`.
- * `marathonProtocol`: The Marathon protocol (`http` or `https`). Default is `http`. 
+ * `marathonProtocol`: The Marathon protocol (`http` or `https`). Default is `http`.
  * `marathonUri`: The relative path where the Marathon Event Bus endpoint can be found. Default is `/v2/events`.
+ * `marathonHeaders`: If you are using Marathon from outside you can add options to the request. Default is an empty object `{}`.
```
Comment (tuxity) [inline README.md]:
```
* `marathonHeaders`: Allows you to add headers to Marathon's API requests. Default is an empty object `{}`.
```

File: README.md [hunk 4]
```diff
@@ -63,8 +63,10 @@ You can specify the following properties when instantiating the Marathon Event B
...
+ Example: `marathonHeaders = {'Authorization': 'token=MARATHON_AUTH_TOKEN', 'Content-Type': 'application/json'}`
```
Comment (tuxity) [inline README.md]:
```
Example: `marathonHeaders = {'Authorization': 'token=DCOS_AUTH_TOKEN', 'Content-Type': 'application/json'}`
```

File: README.md [hunk 5]
```diff
@@ -109,6 +111,7 @@ const eventTypes = ["deployment_info", "deployment_success", "deployment_failed"
 const mebc = new MarathonEventBusClient({
     marathonHost: "localhost", // Use SSE test server
     eventTypes: eventTypes,
+    marathonHeaders: {'Authorization': 'token=MARATHON_AUTH_TOKEN'} // if you are using the api outisde the cluster
```
Comment (tuxity) [inline README.md]: `DCOS_AUTH_TOKEN`

File: examples/example.js [hunk 6]
```diff
@@ -19,6 +19,7 @@ const request = require("request");
 const mebc = new MarathonEventBusClient({
     marathonHost: "localhost", // Use SSE test server
     eventTypes: eventTypes,
+    marathonHeaders: { 'Authorization': 'token=MARATHON_AUTH_TOKEN'}, // if you are using the api outisde the cluster
```
Comment (tuxity) [inline examples/example.js]: `DCOS_AUTH_TOKEN`

File: lib/MarathonEventBusClient.js [hunk 7]
```diff
@@ -30,7 +36,12 @@ function MarathonEventBusClient (options) {
     self.options.marathonPort = parseInt(options.marathonPort) || 8080;
     self.options.marathonProtocol = options.marathonProtocol || "http";
     self.options.marathonUri = options.marathonUri || "/v2/events";
-    
+    /*
+     You can add additionals headers
+     {'Authorization': 'token=MARATHON_AUTH_TOKEN', 'Content-Type': 'application/json'}
```
Comment (tuxity) [inline lib/MarathonEventBusClient.js]: Same

**Review by tuxity — CHANGES_REQUESTED** [review]

File: lib/MarathonEventBusClient.js [hunk 8]
```diff
@@ -30,7 +36,12 @@ function MarathonEventBusClient (options) {
     self.options.marathonPort = parseInt(options.marathonPort) || 8080;
     self.options.marathonProtocol = options.marathonProtocol || "http";
     self.options.marathonUri = options.marathonUri || "/v2/events";
-    
+    /*
+     You can add additionals headers
+     {'Authorization': 'token=API_ACCESS_TOKEN', 'Content-Type': 'application/json'}
+    */
+    self.options.marathonHeaders = options.marathonHeaders || {};
```
Comment (tobilg) [inline lib/MarathonEventBusClient.js]: I think this should be removed, because L60 and after

Comment (tobilg) [reply lib/MarathonEventBusClient.js]: I think it should be

```javascript
let eventSourceInitDict = {
    headers: self.options.marathonHeaders,
    rejectUnauthorized: false
};
```

Please test accordingly. Thanks.

**Review by tobilg — APPROVED** [review]

**Comment by tobilg** [comment]
Merged. I'll release a new version on npm today. Thanks @owndann @tuxity!

**Comment by chl03ks** [comment]
@tobilg  a pleasure.

**Comment by tobilg** [comment]
Version 0.1.5 has been released on npm.

**Pull request merged**